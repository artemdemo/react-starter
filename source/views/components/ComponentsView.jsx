/* eslint-disable class-methods-use-this */
import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {faPlug} from '@fortawesome/free-solid-svg-icons';
import Button, { buttonAppearance, buttonSize } from '../../components/Button/Button';
import Select from '../../components/Select/Select';

class ComponentsView extends React.PureComponent {
    constructor(props) {
        super(props);

        this.state = {
            selectItem1: undefined,
        };

        this.selectList = [
            {name: '1', value: '1'},
            {name: '2', value: '2'},
        ];
    }

    renderButtons() {
        return (
            <>
                <h3 className='mt-3'>Buttons</h3>
                <div className='row'>
                    <div className='col'>
                        <Button appearance={buttonAppearance.LIGHT}>Light</Button>&nbsp;
                        <Button appearance={buttonAppearance.DANGER}>Danger</Button>&nbsp;
                    </div>
                    <div className='col'>
                        <Button size={buttonSize.SM}>Primary small</Button>&nbsp;
                        <Button>Primary standard</Button>&nbsp;
                        <Button size={buttonSize.LG}>Primary large</Button>&nbsp;
                    </div>
                </div>
            </>
        );
    }

    renderSelect() {
        return (
            <>
                <h3 className='mt-3'>Select</h3>
                <div className='row'>
                    <div className='col'>
                        <div className='mb-3'>
                            Simple select (not controlled)<br />
                            <Select
                                list={this.selectList}
                            />
                        </div>
                        Simple select, with disabled item (not controlled)<br />
                        <Select
                            list={[
                                {name: '1', value: '1'},
                                {name: '2', value: '2', disabled: true},
                                {name: '3', value: '3'},
                            ]}
                        />
                    </div>
                    <div className='col'>
                        <div className='mb-3'>
                            With placeholder (not controlled)<br />
                            <Select
                                list={this.selectList}
                                placeholder='Select...'
                            />
                        </div>
                        With placeholder (controlled)<br />
                        <Select
                            list={this.selectList}
                            value={this.state.selectItem1}
                            onChange={(selectItem1) => {
                                this.setState({
                                    selectItem1,
                                }, () => {
                                    // eslint-disable-next-line no-console
                                    console.log(this.state.selectItem1);
                                });
                            }}
                            placeholder='Select...'
                        />
                    </div>
                </div>
            </>
        );
    }

    render() {
        return (
            <>
                <div className='mb-3'>
                    <FontAwesomeIcon icon={faPlug} />
                    &nbsp;
                    Components View
                </div>
                <div className='mb-3'>

                    {this.renderButtons()}

                    {this.renderSelect()}
                </div>
            </>
        );

    }
}

export default ComponentsView;
