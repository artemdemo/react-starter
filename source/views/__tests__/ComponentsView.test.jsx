import React from 'react';
import renderer from 'react-test-renderer';
import ComponentsView from '../components/ComponentsView';

jest.mock('@fortawesome/react-fontawesome');
jest.mock('../../components/Button/Button');
jest.mock('../../components/Select/Select');

describe('<ComponentsView>', () => {
    it('default render', () => {
        const tree = renderer.create(
            <ComponentsView />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
