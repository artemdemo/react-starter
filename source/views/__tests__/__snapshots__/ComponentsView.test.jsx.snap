// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComponentsView> default render 1`] = `
Array [
  <div
    className="mb-3"
  >
    <div
      data-mock="FontAwesomeIcon"
    >
      {
  "icon": {
    "prefix": "fas",
    "iconName": "plug",
    "icon": [
      384,
      512,
      [],
      "f1e6",
      "M320,32a32,32,0,0,0-64,0v96h64Zm48,128H16A16,16,0,0,0,0,176v32a16,16,0,0,0,16,16H32v32A160.07,160.07,0,0,0,160,412.8V512h64V412.8A160.07,160.07,0,0,0,352,256V224h16a16,16,0,0,0,16-16V176A16,16,0,0,0,368,160ZM128,32a32,32,0,0,0-64,0v96h64Z"
    ]
  }
}
    </div>
      Components View
  </div>,
  <div
    className="mb-3"
  >
    <h3
      className="mt-3"
    >
      Buttons
    </h3>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          data-mock="Button"
        >
          {
  "appearance": "light"
}
          Light
        </div>
         
        <div
          data-mock="Button"
        >
          {
  "appearance": "danger"
}
          Danger
        </div>
         
      </div>
      <div
        className="col"
      >
        <div
          data-mock="Button"
        >
          {
  "size": "sm"
}
          Primary small
        </div>
         
        <div
          data-mock="Button"
        >
          {}
          Primary standard
        </div>
         
        <div
          data-mock="Button"
        >
          {
  "size": "lg"
}
          Primary large
        </div>
         
      </div>
    </div>
    <h3
      className="mt-3"
    >
      Select
    </h3>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <div
          className="mb-3"
        >
          Simple select (not controlled)
          <br />
          <div
            data-mock="Select"
          >
            {
  "list": [
    {
      "name": "1",
      "value": "1"
    },
    {
      "name": "2",
      "value": "2"
    }
  ]
}
          </div>
        </div>
        Simple select, with disabled item (not controlled)
        <br />
        <div
          data-mock="Select"
        >
          {
  "list": [
    {
      "name": "1",
      "value": "1"
    },
    {
      "name": "2",
      "value": "2",
      "disabled": true
    },
    {
      "name": "3",
      "value": "3"
    }
  ]
}
        </div>
      </div>
      <div
        className="col"
      >
        <div
          className="mb-3"
        >
          With placeholder (not controlled)
          <br />
          <div
            data-mock="Select"
          >
            {
  "list": [
    {
      "name": "1",
      "value": "1"
    },
    {
      "name": "2",
      "value": "2"
    }
  ],
  "placeholder": "Select..."
}
          </div>
        </div>
        With placeholder (controlled)
        <br />
        <div
          data-mock="Select"
        >
          {
  "list": [
    {
      "name": "1",
      "value": "1"
    },
    {
      "name": "2",
      "value": "2"
    }
  ],
  "placeholder": "Select..."
}
        </div>
      </div>
    </div>
  </div>,
]
`;
